lib/memo_wise/internal_api.rb

```diff
@@ -161,15 +161,70 @@ def self.key_str(method)
     def self.original_class_from_singleton(klass)
       raise ArgumentError, "Must be a singleton class: #{klass.inspect}" unless klass.singleton_class?
 
+      # Since we call this method a lot, we memoize the results.
+      @original_class_from_singleton ||= {}
+
       # Search ObjectSpace
       #   * 1:1 relationship of singleton class to original class is documented
       #   * Performance concern: searches all Class objects
-      #     But, only runs at load time
-      ObjectSpace.each_object(Module).find do |cls|
+      #     But, only runs at load time and results are memoized
+      @original_class_from_singleton[klass] ||= ObjectSpace.each_object(Module).find do |cls|
```

Comment:
To what extent is this additional state (which lives forever inside the `MemoWise` module) justified by quantifiable performance numbers?

Reply:
Great question! It's possible that real-world behavior will typically be less noticeable, but this has a huge impact in our tests. With this change:

```
Finished in 3.53 seconds (files took 0.51088 seconds to load)
438 examples, 0 failures
```

Without this memoization:

```
Finished in 5 minutes 25 seconds (files took 0.41548 seconds to load)
438 examples, 0 failures
```

Should I make that clearer in the comment?

Reply:
Yeah, I think perhaps in the comment would be helpful?